src/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs

```diff
@@ -222,10 +210,13 @@ protected T WithOptions(ApplicationOptions applicationOptions)
             WithClientId(applicationOptions.ClientId);
             WithRedirectUri(applicationOptions.RedirectUri);
             WithTenantId(applicationOptions.TenantId);
-            WithLoggingLevel(applicationOptions.LogLevel);
             WithComponent(applicationOptions.Component);
-            WithEnablePiiLogging(applicationOptions.EnablePiiLogging);
-            WithDefaultPlatformLoggingEnabled(applicationOptions.IsDefaultPlatformLoggingEnabled);
+
+            WithLogging(
+                null, 
+                applicationOptions.LogLevel, 
+                applicationOptions.EnablePiiLogging,
+                applicationOptions.IsDefaultPlatformLoggingEnabled);
```

Comment:
alternatively, why not allow the log callback to be set from the options? (sure, not possible from a file, but it could be set using code...)

---
In reply to: [252130323](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/851#discussion_r252130323) [](ancestors = 252130323)

Reply:
Yes, we want to enable setting loglevel/pii/platform from the app config / options serialization.  Sicne the options is meant to be deserialized, we can't put the callback function there.  So the pattern is to do:  CreateWithOptions(options).WithLogging((params) => {do logging})

---
In reply to: [252130663](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/851#discussion_r252130663) [](ancestors = 252130663,252130323)

Reply:
then why not have another WithLogging that doesn't take the delegate?

---
In reply to: [252415270](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/851#discussion_r252415270) [](ancestors = 252415270,252130663,252130323)